core/storage/blog_post/gae_models.py

```diff
@@ -0,0 +1,454 @@
+        """
+        return base_models.MODEL_ASSOCIATION_TO_USER.MULTIPLE_INSTANCES_PER_USER
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model contains data corresponding to a user to export."""
+        return dict(super(BlogPostModel, cls).get_export_policy(), **{
+            # We do not export the author_id because we should not
+            # export internal user ids.
+            'author_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'title': base_models.EXPORT_POLICY.EXPORTED,
+            'content': base_models.EXPORT_POLICY.EXPORTED,
+            'url_fragment': base_models.EXPORT_POLICY.EXPORTED,
+            'tags': base_models.EXPORT_POLICY.EXPORTED,
+            'thumbnail_filename': base_models.EXPORT_POLICY.EXPORTED,
+            'published_on': base_models.EXPORT_POLICY.EXPORTED,
+        })
+
+    @classmethod
+    def generate_new_blog_post_id(cls):
+        """Generates a new blog post ID which is unique and is in the form of
+        random hash of 12 chars.
+
+        Returns:
+            str. A blog post ID that is different from the IDs of all
+            the existing blog posts.
+
+        Raises:
+            Exception. There were too many collisions with existing blog post
+                IDs when attempting to generate a new blog post ID.
+        """
+
+        for _ in python_utils.RANGE(base_models.MAX_RETRIES):
+            blog_post_id = utils.convert_to_hash(
+                python_utils.UNICODE(
+                    utils.get_random_int(base_models.RAND_RANGE)),
+                base_models.ID_LENGTH)
+            if not cls.get_by_id(blog_post_id):
+                return blog_post_id
+        raise Exception(
+            'New blog post id generator is producing too many collisions.')
+
+    @classmethod
+    def create(cls, blog_post_id, author_id):
+        """Creates a new BlogPostModel entry.
+
+        Args:
+            blog_post_id: str. Blog Post ID of the newly-created blog post.
+            author_id: str. User ID of the author.
+
+        Returns:
+            BlogPostModel. The newly created BlogPostModel instance.
+
+        Raises:
+            Exception. A blog post with the given blog post ID exists already.
+        """
+        if cls.get_by_id(blog_post_id):
+            raise Exception(
+                'A blog post with the given blog post ID exists already.')
+
+        entity = cls(
+            id=blog_post_id,
+            author_id=author_id,
+            content='',
+            title='',
+            published_on=None,
+            url_fragment='',
+            tags=[],
+            thumbnail_filename=''
+        )
+        entity.update_timestamps()
+        entity.put()
+
+        return entity
+
+    # If the user directly enters the blog post's url in the editor or the
+    # homepage, the blogPostModel will be queried using the url fragment to
+    # retrieve data for populating the editor dashboard / blog post page.
```

Comment:
Please move this to the documentation above `url_fragment = StringProperty(...)`

Reply:
Done.